crates/alpen-ee/sequencer/src/block_builder/task.rs

```diff
@@ -124,21 +141,20 @@ pub async fn block_builder_task<
 }
 
 async fn block_builder_task_inner<TEngine: PayloadBuilderEngine>(
+    next_block_target: &BlockTarget,
     config: &BlockBuilderConfig,
     exec_chain_handle: &ExecChainHandle,
     ol_chain_handle: &OLChainTrackerHandle,
     payload_builder: &TEngine,
     storage: &impl ExecBlockStorage,
     clock: &impl Clock,
```

Comment:
Can some of this be bundled up and hidden behind some `BuilderContext`/`BuilderContextProvider` trait?